cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizerInstantiatorService.java

```diff
@@ -17,277 +17,74 @@
-    this.authorizationEnabled = cConf.getBoolean(Constants.Security.Authorization.ENABLED);
-    this.instantiatorFactory = new InstantiatorFactory(false);
     this.authorizationContextFactory = authorizationContextFactory;
   }
 
-  @Override
-  protected void startUp() throws Exception {
-    if (!authorizationEnabled) {
-      LOG.debug("Authorization is disabled. Using a no-op authorizer.");
-      this.authorizer = new NoOpAuthorizer();
-      return;
-    }
-    if (!authenticationEnabled) {
-      LOG.debug("Authorization is enabled. However, authentication is disabled. Using a no-op authorizer.");
-      this.authorizer = new NoOpAuthorizer();
-      return;
-    }
-    // Authorization is enabled, so continue with startup now
-    String authorizerExtensionJarPath = cConf.get(Constants.Security.Authorization.EXTENSION_JAR_PATH);
-    if (Strings.isNullOrEmpty(authorizerExtensionJarPath)) {
-      throw new IllegalArgumentException(
-        String.format("Authorizer extension jar path not found in configuration. Please set %s in cdap-site.xml to " +
-                        "the fully qualified path of the jar file to use as the authorization backend.",
-                      Constants.Security.Authorization.EXTENSION_JAR_PATH));
-    }
-    File authorizerExtensionJar = new File(authorizerExtensionJarPath);
-    ensureValidAuthExtensionJar(authorizerExtensionJar);
-    File tmpDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
-                           cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
-    this.tmpDir = DirUtils.createTempDir(tmpDir);
-    this.authorizerClassLoader = createAuthorizerClassLoader(authorizerExtensionJar);
-    this.authorizer = createAndInitializeAuthorizerInstance(authorizerExtensionJar);
-  }
-
-  @Override
-  protected void shutDown() throws Exception {
-    authorizer.destroy();
-    if (!authorizationEnabled || !authenticationEnabled) {
-      // nothing to close, since we would not have created a class loader
-      return;
-    }
-    try {
-      authorizerClassLoader.close();
-    } catch (IOException e) {
-      LOG.warn("Failed to close authorizer class loader", e);
-    }
-    try {
-      DirUtils.deleteDirectoryContents(tmpDir);
-    } catch (IOException e) {
-      // It's a cleanup step. Nothing much can be done if cleanup fails.
-      LOG.warn("Failed to delete directory {}", tmpDir, e);
-    }
-  }
-
   /**
-   * Returns an instance of the configured {@link Authorizer} extension, or of {@link NoOpAuthorizer}, if
-   * authorization is disabled.
+   * Initiates the startup of the underlying {@link AuthorizationService}.
    */
-  @Override
-  public Authorizer get() {
-    Preconditions.checkState(isRunning(), "Authorization Service has not yet started. Authorizer not available.");
-    return authorizer;
+  public ListenableFuture<Service.State> start() {
+    delegate = new AuthorizationService(cConf, authorizationContextFactory);
```

Comment:
to be safe, if ``delegate`` is not null, then should we check its state before recreating it?